packages/webapp/src/store/api/apiSlice.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { RootState } from '../store';
+import { animalsUrl, animalBatchesUrl, animalGroupsUrl, url } from '../../apiConfig';
+import type { Animal, AnimalBatch, AnimalGroup } from './types';
+
+export const api = createApi({
```

Comment:
As mentioned on Antos PR I think this file is going to get huge. I know the redux docs have this pattern for their examples but in the class I took they did it like this:

```
store 
> apis 
    > animalBatchesApi 
    > animalApi
> slices
    > whateverSlice
> thunks
    > whateverThunk
```

This has the benefit of setting the baseQuery to match the endpoints, we can start using the new pattern where we don't have to pass in user_id to the headers for the other routes. And of course the nice pattern of making an redux api for each new table.

Reply:
I think Anto said we will start with one file but almost certainlly move to some sort of chunking. But looking at this closer, I'm not sure about one file-one endpoint (is that what the diagram is indicating?) I guess I would have thought of chunking all the animal endpoints together, although now I realize our API wasn't really set up like that with a common base URL 🤔

Reply:
Even all the animal endpoints together could get chunky -- image addOne, removeOne, addMany, fetchOne, fetchMany etc etc. for each of animal, animal batches, animal groups, animal identifiers, animal sexes, breeds, types etc. etc.

With one per endpoint it would mirror our routes/models/tables with probably one clearly defined get, post, and delete and typings for most.

Reply:
True, true. And yes it does feel particularly appealing to mirror the routes file exactly...!